core/string.rbs

```diff
@@ -1349,7 +1349,7 @@ class String
   # modification made, `self` otherwise.
   #
   def chomp!: (nil) -> nil
-            | (?string? separator) -> self?
```

Comment:
This case is necessary.
Passing a node of type `String?` will be a type error without the `?`.

Reply:
This is a problem with the type checkers unable to break `string` and `nil` from `string?` to different paths and then merge them back to `self? | nil` i.e. `self?`

Reply:
It may be true. Type checkers can do more about this.
But at least for now, RBS is for static type checking and having `string?` is more common for static type checking (I believe.)

Reply:
> But at least for now

How about this: I will revert it **for now**, but we should also remind type checker maintainers about these sort of cases.

Reply:
soutaro/steep#333